jetty-server/src/main/java/org/eclipse/jetty/server/Response.java

```diff
@@ -506,7 +506,9 @@ public void sendRedirect(int code, String location) throws IOException
 
         if (!URIUtil.hasScheme(location))
         {
-            StringBuilder buf = _channel.getRequest().getRootURL();
+            StringBuilder buf = _channel.getHttpConfiguration().isRelativeRedirectionAllowed()
+                ? new StringBuilder()
+                : _channel.getRequest().getRootURL();
```

Comment:
Is the Servlet specification mandating absolute redirects? Just to be sure that we actually can.

Reply:
probably, but it was written before RFC7230 and we do absolute by default.  Happy to bend the rules if users configure it to do so.